arch/risc-v/src/esp32c3/esp32c3_rtc_lowerhalf.c

```diff
@@ -0,0 +1,575 @@
+      goto errout;
+    }
+
+  return OK;
+
+errout:
+  DEBUGASSERT(ret < 0);
+  return ret;
+
+#else
+  time_t timer;
+
+  /* The resolution of time is only 1 second */
+
+  timer = up_rtc_time();
+
+  /* Convert the one second epoch time to a struct tm */
+
+  if (gmtime_r(&timer, (FAR struct tm *)rtctime) == 0)
+    {
+      int errcode = get_errno();
+      DEBUGASSERT(errcode > 0);
+
+      rtcerr("ERROR: gmtime_r failed: %d\n", errcode);
+      return -errcode;
+    }
+
+  return OK;
+#endif
+}
+
+/****************************************************************************
+ * Name: rtc_settime
+ *
+ * Description:
+ *   Implements the settime() method of the RTC driver interface
+ *
+ * Input Parameters:
+ *   lower   - A reference to RTC lower half driver state structure
+ *   rcttime - The new time to set
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on any failure.
+ *
+ ****************************************************************************/
+
+static int rtc_settime(FAR struct rtc_lowerhalf_s *lower,
+                       FAR const struct rtc_time *rtctime)
+{
+  struct timespec ts;
+
+  /* Convert the struct rtc_time to a time_t.  Here we assume that struct
+   * rtc_time is cast compatible with struct tm.
+   */
+
+  ts.tv_sec  = mktime((FAR struct tm *)rtctime);
+  ts.tv_nsec = 0;
+
+  /* Now set the time (to one second accuracy) */
+
+  return up_rtc_settime(&ts);
+}
+
+/****************************************************************************
+ * Name: rtc_havesettime
+ *
+ * Description:
+ *   Implements the havesettime() method of the RTC driver interface
+ *
+ * Input Parameters:
+ *   lower - A reference to RTC lower half driver state structure
+ *
+ * Returned Value:
+ *   Returns true if RTC date-time have been previously set.
+ *
+ ****************************************************************************/
+
+static bool rtc_havesettime(FAR struct rtc_lowerhalf_s *lower)
+{
+  if (esp32c3_rtc_get_boot_time() == 0)
+    {
+      return false;
+    }
+
+  return true;
+}
+
+/****************************************************************************
+ * Name: rtc_setalarm
+ *
+ * Description:
+ *   Set a new alarm. This function implements the setalarm() method of the
+ *   RTC driver interface
+ *
+ * Input Parameters:
+ *   lower - A reference to RTC lower half driver state structure
+ *   alarminfo - Provided information needed to set the alarm
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on any failure.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+static int rtc_setalarm(FAR struct rtc_lowerhalf_s *lower,
+                        FAR const struct lower_setalarm_s *alarminfo)
+{
+  FAR struct esp32c3_lowerhalf_s *priv;
+  FAR struct esp32c3_cbinfo_s *cbinfo;
+  struct alm_setalarm_s lowerinfo;
+  int ret = -EINVAL;
```

Comment:
No need to initialize.
It will be overwritten in line 364

Reply:
Done